web/packages/plugins/shared/common/src/utils/truncatePathString.ts

```diff
@@ -0,0 +1,30 @@
+export const truncatePathString = (path: string): string => {
+  /*
+  Truncate path string to have length <= MAX_LENGTH.
+   */
+  const MAX_LENGTH: number = 55
+  if (!path.includes('/') || path.length <= MAX_LENGTH) {
+    return path
+  }
+  const splitTextOnFolders: string[] = path.split('/')
+  const firstFolder: string = splitTextOnFolders[0]
+  const secondFolder: string = splitTextOnFolders[1]
+  const finalName: string = splitTextOnFolders[splitTextOnFolders.length - 1]
+  const truncatedText: string = `${firstFolder}/${secondFolder}/.../${finalName}`
+  const truncatedTextSmall: string = `${firstFolder}/.../${finalName}`
+
+  if (truncatedText.length <= MAX_LENGTH) {
+    return truncatedText
+  } else if (
+    truncatedText.length > MAX_LENGTH &&
+    truncatedTextSmall.length <= MAX_LENGTH
+  ) {
+    return truncatedTextSmall
```

Comment:
This is nice.
But would it be possible to add one and one part, alternating from start and end, until the "MAX_LENGTH" is reached?
This would give the most possible information that can fit inside the input field :)

Reply:
done!